entry/encoder-entry

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+
+SELF=$(basename $0)
+
+function err() {
+  echo "ERR $SELF $$ ${FUNCNAME[1]}: $@" >&2
+}
+
+function dbg() {
+  echo "DBG $SELF $$ ${FUNCNAME[1]}: $@" >&2
+}
+
+function help() {
+    cat <<EOF
+Usage: $SELF [-i <id>] [-u <uid>] [-h] <url>
+  Streamer provide streaming service between AIC and WebRTC client
+
+  -i <id>: instance ID, default: $id
+  -u <uid>: user id, default: 0
+  -g <gop_size>: gop size, default: $gop
+  -h:      print the usage message
+  <ip>:    ip to signaling server in format
+
+EOF
+}
+
+function main() {
+  local uid=0
+  local gop=0
+  local user_count=${K8S_ENV_USER_COUNT_PER_ANDROID:-1}
+
+  while getopts ":i:u:g:h" opt; do
+    case $opt in
+    i) id=$OPTARG ;;
+    u) uid=$OPTARG;;
+    g) gop=$OPTARG ;;
+    h) help && exit 1 ;;
+    *) echo "no such option: $opt" && exit 1 ;;
+    esac
+  done
+  shift $((OPTIND - 1))
+
+  if [[ -n "$1" ]]; then
+    ip="$1"
+  else
+    help
+    exit 1
```

Comment:
Simplification suggestion:

Add error message argument for `help()` function, and show it before doing `exit 1` at the end of `help()`.  Maybe also rename it to something like `error_exit()` for clarity.  Then above two lines (and `*)` case handling above) would be just `error_exit "argument missing"`.